src/StringTools.UnitTests/OpportunisticIntern_Tests.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET35_UNITTEST
+extern alias StringToolsNet35;
+#endif
+
+using Shouldly;
+using Xunit;
+
+#if NET35_UNITTEST
+using StringToolsNet35::System;
+using StringToolsNet35::Microsoft.StringTools;
+#else
+using System;
+#endif
+
+namespace Microsoft.StringTools.Tests
+{
+    public class OpportunisticInternTestBase
+    {
+        private static bool IsInternable(ref InternableString internable)
+        {
+            string i1 = OpportunisticIntern.Instance.InternableToString(ref internable);
+            string i2 = OpportunisticIntern.Instance.InternableToString(ref internable);
+            i1.ShouldBe(i2); // No matter what, the same string value should return.
+            return System.Object.ReferenceEquals(i1, i2);
+        }
+
+        private static void AssertInternable(ref InternableString internable)
+        {
+            IsInternable(ref internable).ShouldBeTrue();
+        }
+
+        private static string AssertInternable(char[] ch, int startIndex, int count)
+        {
+            var target = new InternableString(ch.AsSpan(startIndex, count));
+            AssertInternable(ref target);
+            target.Length.ShouldBe(count);
+
+            return target.ExpensiveConvertToString();
+        }
+
+        private static void AssertInternable(string value)
+        {
+            AssertInternable(value.ToCharArray(), 0, value.ToCharArray().Length);
```

Comment:
Maybe add a check that the return value equals `value`?

Reply:
Sure, done!